pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
Let me see if I understand. With the suggested change, `alpha.kubernetes.io/*`, `beta.kubernetes.io/*`, `kubernetes.io/*`, and any resource not of the form `xxx/*` will be considered a first class resource. All resources of the form `xxx/*` that does not match `alpha.kubernetes.io/*`, `beta.kubernetes.io/*`, or `kubernetes.io/*` will be considered opaque integer resources (or extended resources, which I admit I haven't read up on, so I don't really know what the desired semantics are for that). Is that correct?

If so, it seems fine to me, assuming it doesn't break any existing resources.